### PR TITLE
Docker dev environment

### PR DESCRIPTION
**Pull request by rimbi**
This will provide a standardized development environment as well as the necessary infrastructure for the debian package management.

### CODE REVIEW

File: .cargo/config [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[target.x86_64-unknown-linux-gnu]
+linker = "x86_64-linux-gnu-gcc"
```
Comment (zeenix) [inline .cargo/config:2]: I don't get it. You're specifying linker for x86 target, how is that relevant to arm cross built? :thinking:

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	// "forwardPorts": [],
+
+	// Uncomment the next line to run commands after the container is created.
+	// "postCreateCommand": "cat /etc/os-release",
+
+	// Configure tool-specific properties.
+	// "customizations": {},
+
+	// Uncomment to connect as an existing user other than the container default. More info: https://aka.ms/dev-containers-non-root.
+	// "remoteUser": "devcontainer"
```
Comment (zeenix) [inline .devcontainer/devcontainer.json:25]: We don't need these. If people need to add them, they can consult the docs you point to above and add them themselves.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM rust
+
+RUN cargo install cargo-watch cargo-deb
```
Comment (zeenix) [inline Dockerfile:3]: I wouldn't provide any distro-specific stuff.

**Review by zeenix — CHANGES_REQUESTED** [review]
Would be good to

* add instructions in the README on how to use this.
* CI to test image build and building with image.

File: .cargo/config (hunk 1 above)
Comment (rimbi) [reply .cargo/config:2]: x86 is the target for the cross compilation.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (rimbi) [reply .devcontainer/devcontainer.json:25]: I would prefer other way but no problem. I can add `.devcaontainer` folder to `.gitignore`.

File: Dockerfile (hunk 3 above)
Comment (rimbi) [reply Dockerfile:3]: `cargo-deb` is necessary to build the `deb` packages, which might be good to have it accessible.

**Comment by rimbi** [comment]
Initially, I would like to use it just with VsCode. No other purposes. When you open a project with a Docker file, VsCode prompts the option for opening the project in the container and then builds and prepares the container env.

File: Dockerfile (hunk 3 above)
Comment (zeenix) [reply Dockerfile:3]: Yeah I know what it would be for but it's distro-specific and hence not something I'd want here.

Is building deb natively harder?

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (zeenix) [reply .devcontainer/devcontainer.json:25]: You misunderstand. I was referring to the commented out settings, not the file itself.

File: .cargo/config (hunk 1 above)
Comment (zeenix) [reply .cargo/config:2]: ah ok. Thing is that this will be used on normal native builds too and that would not be good. E.g I use mold linker for faster linking and AFAICT, this would override that.

**Comment by zeenix** [comment]
> Initially, I would like to use it just with VsCode. No other purposes. When you open a project with a Docker file, VsCode prompts the option for opening the project in the container and then builds and prepares the container env.

Nice! However, given that this project is quite small and has very few deps (which are fetched by cargo for you) and I don't expect a lot of developers to get involved, I'd only want this nice workflow added if it can be done in a very non-intrusive (e.g no changes to default linker on x86 Linux) manner.

With #27, I was more thinking of users, not developers.

File: Dockerfile (hunk 3 above)
Comment (rimbi) [reply Dockerfile:3]: What do you mean with `natively`? Without using `cargo-deb`?

**Comment by rimbi** [comment]
Ok, I am closing this PR as I don't have enough energy to continue the discussion. :)

File: Dockerfile (hunk 3 above)
Comment (zeenix) [reply Dockerfile:3]: I mean w/o using docker or any containers, like I normal develop. :)

**Comment by zeenix** [comment]
> Ok, I am closing this PR as I don't have enough energy to continue the discussion. :)

Fair enough. I hope no hard feelings though and that you understand me disagreeing doesn't meant that I do not appreciate your contributions (even ones that don't get merged).

**Comment by rimbi** [comment]
Not at all. Apparently you don't need much help here and can move faster alone. Please don't hesitate to contact me when you need help.
